fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -1107,6 +1106,16 @@ boolean performIndexingStampOperation(@Nonnull ConcurrentHashMap<String, IndexBu
         return true;
     }
 
+    public CompletableFuture<Map<UUID, IndexBuildProto.IndexBuildHeartbeat>> getIndexingHeartbeats(int maxCount) {
+        return getRunner().runAsync(context -> openRecordStore(context)
+                        .thenCompose(store -> IndexingHeartbeat.getIndexingHeartbeats(store, common.getPrimaryIndex(), maxCount)));
+    }
+
+    public CompletableFuture<Integer> clearIndexingHeartbeats(long minAgenMilliseconds, int maxIteration) {
```

Comment:
Should this use `ThrottledRetryingIterator`, removing the need for `maxIteration`?

Reply:
IDK. The `maxIteration` is not expected to be used unless something went wrong. Very wrong. Using the `ThrottledRetryingIterator` might be an overkill.